tests/unit/test_charm.py

```diff
@@ -579,6 +600,54 @@ def test_team_map():
     )
 
 
+def test_swift_config(
+    harness: ops.testing.Harness,
+    setup_replica_consensus: typing.Callable[[], dict],
+):
+    """
+    arrange: after peer relation established and database ready
+    act: update legacy version of the wp_plugin_openstack-objectstorage_config configuration
+    assert: parsed swift configuration should update all legacy fields.
+    """
+    harness.set_can_connect(harness.model.unit.containers["wordpress"], True)
+    setup_replica_consensus()
+    harness.update_config(
+        {
+            "wp_plugin_openstack-objectstorage_config": json.dumps(
+                {
+                    "auth-url": "http://swift.test/identity/v3",
+                    "bucket": "wordpress_tests.integration.test_upgrade",
+                    "password": "nomoresecret",
+                    "region": "RegionOne",
+                    "tenant": "demo",
+                    "domain": "default",
+                    "username": "demo",
+                    "copy-to-swift": "1",
+                    "serve-from-swift": "1",
+                    "remove-local-file": "0",
+                    "url": "http://swift.test:8080/v1/AUTH_fa8326b9fd4f405fb1c5eaafe988f5fd/wordpress_tests.integration.test_upgrade/wp-content/uploads/",
+                    "prefix": "wp-content/uploads/",
+                }
+            )
+        }
+    )
+    charm: WordpressCharm = typing.cast(WordpressCharm, harness.charm)
+    assert charm._swift_config() == {
+        "auth-url": "http://swift.test/identity/v3",
```

Comment:
Would it be worth it to define a variable to hold this configuration so that it doesn't have to be defined twice?

Reply:
The swift configuration above is actually different from this one. Notice the `url` and `prefix`.

Reply:
ah, so maybe those things would just be updated before the comparison? That would make it clearer what the difference is as well

Reply:
Updated, thanks!